src/Microsoft.SymbolStore/KeyGenerators/ELFFileKeyGenerator.cs

```diff
@@ -196,7 +196,7 @@ private string GetSymbolFileName()
         /// <returns>symbol store keys</returns>
         private static void NormalizeBuildId(ref byte[] buildId)
         {
-            if (buildId != null && buildId.Length < 20 && buildId.Length >= 16)
```

Comment:
This should perhaps throw an exception if buildID is greater than 20 bytes (and not sure why we don't do that in null cases), and remove the 8 byte check to be compliant with the spec.